shaping/wrapping.go

```diff
@@ -362,6 +362,97 @@ func (r *runMapper) mapRun(runIdx int, run Output) {
 	}
 }
 
+// RunIterator defines a type that can incrementally provide shaped text.
+type RunIterator interface {
+	// Next returns the next logical run index, shaped run, and true if there are more runs,
+	// then it advances the iterator to the next element.
+	// Otherwise it returns an undefined index, an empty Output, and false if there are no more outputs.
```

Comment:
Since you wrote "Otherwise", the returned boolean will be false, won't it ? 
So the "if there are no more outputs" seems a bit redundant/confusing.